src/lib/discord/bot.ts

```diff
@@ -13,13 +17,261 @@ new Command({
+		name: 'query',
+		description: 'Query to search for',
+		type: ApplicationCommandOptionType.String,
+		required: true,
+	})
+	.handle(async data => {
+		console.log(data)
+		//query should always be a string. probably a better way to do this
+		if (typeof data.options.query !== 'string') throw new Error('Custom ID is invalid')
+		const query: string = data.options.query
+		const entries = await fetchEntries({
+			query: query,
+			limit: 10,
+			skip: 0,
+		})
+		const embed: APIEmbed = {
+			title: `Search Results for ${query}`,
+			fields: [],
+			footer: {
+				text: 'Page 1',
+			},
+		}
+		for (const entry of entries) {
+			embed.fields!.push({
+				name: entry.title,
+				value: `[Link](${process.env.BASE_URL}/entry/${entry.slug})`,
+			})
+		}
+		//note: as of now buttons just sit around. clicking on them does not do anything
+		return {
+			components: [
+				{
+					//buttons must be in an action row.... why? blame discord
+					type: 1,
+					components: [
+						{
+							//type 2 = button
+							type: 2,
+							style: 1,
+							label: 'Back',
+							emoji: {
+								id: null,
+								name: '◀️',
+							},
+							//command-button-page-query
+							custom_id: `search-back-1-${query}`,
+						},
+						{
+							type: 2,
+							style: 1,
+							label: 'Forward',
+							emoji: {
+								id: null,
+								name: '▶️',
+							},
+							custom_id: `search-forward-1-${query}`,
+						},
+					],
+				},
+			],
+			embeds: [embed],
+		}
+	})
+	.handleComponents(async args => {
+		const action = args[0]
+		//`+` is unary operator that converts to int
+		let page: number = +args[1]
+		const query = args[2]
+		let entries = await fetchEntries({
+			query: query,
+			limit: 0,
+			skip: (page - 1) * 10,
+		})
+		entries = entries.slice(0, 10)
+		if (action == 'back') {
+			page--
+			if (page == 0) {
+				page = 1
+			}
+		} else if (action == 'forward') {
+			const max_pages = Math.ceil(entries.length / 10)
+			if (!(page + 1 > max_pages)) {
+				page++
+			}
+		}
+		const embed: APIEmbed = {
+			title: `Search Results for ${query}`,
+			fields: [],
+			footer: {
+				text: `Page ${String(page)}`,
+			},
+		}
+		for (const entry of entries) {
+			embed.fields!.push({
+				name: entry.title,
+				value: `[Link](${process.env.BASE_URL}/entry/${entry.slug})`,
+			})
+		}
+		return {
+			components: [
+				{
+					//buttons must be in an action row.... why? blame discord
+					type: 1,
+					components: [
+						{
+							//type 2 = button
+							type: 2,
+							style: 1,
+							label: 'Back',
+							emoji: {
+								id: null,
+								name: '◀️',
+							},
+							//command-button-page-query
+							custom_id: `search-back-${String(page)}-${query}`,
+						},
+						{
+							type: 2,
+							style: 1,
+							label: 'Forward',
+							emoji: {
+								id: null,
+								name: '▶️',
+							},
+							custom_id: `search-format-${String(page)}-${query}`,
+						},
+					],
+				},
+			],
+			embeds: [embed],
+		}
+	})
```

Comment:
This code is partially duplicated, there should just be a function that generates the embed.

Reply:
I don't believe that this will be possible without causing more issues with Discord interactions

Reply:
too hard to do this. maybe next time

Reply:
I will be setting this conversation to partially resolved!